redaxo/src/core/lib/setup/setup.php

```diff
@@ -260,6 +260,78 @@ public static function isInitialSetup(): bool
         }
     }
 
+    /**
+     * @return string|false Setup URL or `false` on failure
+     */
+    public static function startWithToken()
+    {
+        $token = rtrim(strtr(base64_encode(random_bytes(32)), '+/', '-_'), '=');
+
+        $configFile = rex_path::coreData('config.yml');
+        $config = rex_file::getConfig($configFile);
+
+        $config['setup'] = isset($config['setup']) && is_array($config['setup']) ? $config['setup'] : [];
+        $config['setup'][$token] = (new DateTimeImmutable('+1 hour'))->format('Y-m-d H:i:s');
+
+        if (false === rex_file::putConfig($configFile, $config)) {
+            return false;
+        }
+
+        return rex_url::backendPage('setup', ['setup_token' => $token], false);
+    }
+
+    public static function isEnabled(): bool
+    {
+        $setup = rex::getProperty('setup', false);
+
+        if (!is_array($setup)) {
+            return (bool) $setup;
```

Comment:
```suggestion
// system wide setup
```

Reply:
Irgendwie fehlen in deinen Suggestions die Eindrückungen. Werde die daher händisch übernehmen.